### PR TITLE
fix(cache): include toolchain versions in optimum compiled model cache key

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

`_generate_model_path_name` computes the cache key (directory name under compiled_models/) for a compiled model. Until now it hashed only user-provided parameters (model, batch size, block size, max len, TP size, rbln_config), but not the versions of the toolchain that actually produces the binary.

This meant that after upgrading `rebel-compiler` or `optimum-rbln`, the same key could resolve to a stale cache entry compiled by a different toolchain — a silent cache collision. This was already flagged by a FIXME in the code.

This PR folds the installed `rebel-compiler` / `optimum-rbln` versions into the hash, so a toolchain upgrade naturally invalidates the cache and triggers a recompile.

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [x] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/utils/optimum/converter/dispatch.py [hunk 1]
```diff
@@ -53,6 +54,31 @@ def _strip_runtime_only_keys(obj: dict) -> dict:
     return obj
 
 
+# Packages whose version changes the compiled binary, so a version bump must
+# invalidate the cache to avoid loading an artifact built by a different
+# toolchain.
```
Comment (rebel-jonghewk) [inline vllm_rbln/utils/optimum/converter/dispatch.py:59]: Why is recompilation necessary here? Different versions may still be compatible.
